src/main/java/org/spongepowered/api/entity/ai/Goal.java

```diff
@@ -60,19 +60,20 @@
     /**
      * Adds a new {@link AITask} to this goal.
      *
-     * @param priority The priority this task should run at
+     * @param priority The priority this task should run at. Smaller numbers
+     *                 indicates higher priority
      * @param task The task to run
      * @return This goal, for chaining
      */
-    Goal<O> addTask(int priority, AITask<? extends O> task);
+    Goal<O> addTask(int priority, AITask<? super O> task);
 
     /**
      * Removes a specific {@link AITask} from this goal.
      *
      * @param task The task to remove
      * @return This goal, for chaining
      */
-    Goal<O> removeTask(AITask<? extends O> task);
+    Goal<O> removeTask(AITask<? super O> task);
```

Comment:
http://stackoverflow.com/questions/4343202/difference-between-super-t-and-extends-t-in-java

As that link says, if there is a generic `Goal<Zombie>` and `AITask<Agent>`, the task should be a valid parameter for method `removeTask`. If the task is `AITask<ZombiePigman>`, then the task should not be a valid parameter.  `AITask<? extends O>` allows you to pass in arguments like `AITask<ZombiePigmen>`, while `AITask<? super O>` allows arguments like `AITaks<Agent>`.